### PR TITLE
handle pdflatex error better

### PR DESCRIPTION
**Pull request by ChristosT**
Correct me if I am wrong but I believe if you use the makefile and fail to compile the document ( I know this cannot happen everybody writes correct latex code at his/her first try :D ) then due to the fact that:
* ['inputAllFiles'](https://github.com/oduwsdl/wsdlthesis/blob/ea0ec87f385ab835240679fe81e252cdf6a874ef/custommacros.sty#L13) appends contents without checking for the existence of the temp file 
* `  \AtEndDocument{\app@exe{rm -f #1/\jobname.tmp}}}` will not be executed.
new contents will keep concatenating upon the original flawed file. So the error remains forever!

TL;DR
executes `\app@exe{rm -f #1/\jobname.tmp}` in `\inputAllFiles` before everything else to ensure a clean initial state.

### CODE REVIEW

File: custommacros.sty [hunk 1]
```diff
@@ -11,6 +11,7 @@
 \makeatletter
 \def\app@exe{\immediate\write18}
 \def\inputAllFiles#1{%
+  \app@exe{rm -f #1/\jobname.tmp}
```
Comment (ibnesayeed) [inline custommacros.sty]:
```suggestion
```

File: custommacros.sty [hunk 2]
```diff
@@ -11,6 +11,7 @@
...
   \app@exe{ls #1/*.tex | xargs cat >> #1/\jobname.tmp}%
```
Comment (ibnesayeed) [inline custommacros.sty]: How about making it simpler which does not rely on `xargs` and append mode, but concatenate everything together and overwrite the temporary file each time?

```suggestion
  \app@exe{cat #1/*.tex > #1/\jobname.tmp}%
```

**Review by ibnesayeed — COMMENTED** [review]
How about this simplified suggestion @ChristosT? Your thoughts?

**Comment by ChristosT** [comment]
even better !

**Comment by ibnesayeed** [comment]
> @ChristosT: even better !

It looks like you only applied one suggestion (that reverts the change) and not the other (which improves on the existing state).

**Comment by ChristosT** [comment]
yes I noticed that but it does not allow me to accept both. marking the second as outdated

**Comment by ibnesayeed** [comment]
Please go ahead and commit that change manually.

**Comment by ChristosT** [comment]
should be ok now

**Comment by ibnesayeed** [comment]
Thanks @ChristosT.
